service/matching/taskQueueManager.go

```diff
@@ -426,16 +453,59 @@ func (c *taskQueueManagerImpl) DispatchQueryTask(
 	return c.matcher.OfferQuery(ctx, task)
 }
 
+// GetVersioningData returns the versioning data for the task queue if any. If this task queue is a sub-partition and
+// has no cached data, it will explicitly attempt a fetch from the root partition.
 func (c *taskQueueManagerImpl) GetVersioningData(ctx context.Context) (*persistencespb.VersioningData, error) {
-	return c.db.GetVersioningData(ctx)
+	vd, err := c.db.GetVersioningData(ctx)
+	if errors.Is(err, &versioningDataNotPresentOnSubPartition{}) {
+		// If this is a sub-partition with no versioning data, this call is indicating we might expect to find
+		// some. Since we may not have started the poller, we want to explicitly attempt to fetch now, because
+		// it's possible some was added to the root partition, but it failed to invalidate this sub partition.
+		return c.fetchMetadataFromRootPartition(ctx)
+	}
+	return vd, err
 }
 
 func (c *taskQueueManagerImpl) MutateVersioningData(ctx context.Context, mutator func(*persistencespb.VersioningData) error) error {
 	err := c.db.MutateVersioningData(ctx, mutator)
+	// If this is the root partition, notify sub partitions that they should fetch changed data from us.
+	if c.taskQueueID.IsRoot() {
+		numParts := c.config.NumReadPartitions()
```

Comment:
oh man, this is where the read/write thing gets messy. it's to support changing partition count, but it's not really tested. for now, can you do max(read, write) here? that should be harmless and not break things any more

Reply:
SGTM